File: src/com/crossbowffs/quotelock/provider/PreferenceProvider.java
package com.crossbowffs.quotelock.provider;

import android.database.Cursor;
import android.net.Uri;

import com.crossbowffs.quotelock.BuildConfig;
import com.crossbowffs.quotelock.consts.PrefKeys;
import com.crossbowffs.remotepreferences.RemotePreferenceProvider;

public class PreferenceProvider extends RemotePreferenceProvider {
    public static final String AUTHORITY = BuildConfig.APPLICATION_ID + ".preferences";

    public PreferenceProvider() {
        super(AUTHORITY, new String[] {PrefKeys.PREF_COMMON, PrefKeys.PREF_QUOTES});
    }

    @Override
    protected boolean checkAccess(String prefName, String prefKey, boolean write) {
        return !write;
    }

    @Override
    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
        return super.query(uri, projection, selection, selectionArgs, sortOrder);
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/provider/PreferenceProvider.kt
package com.crossbowffs.quotelock.provider

import android.content.Context
import android.content.SharedPreferences
import android.database.Cursor
import android.net.Uri
import com.crossbowffs.quotelock.consts.PREF_COMMON
import com.crossbowffs.quotelock.consts.PREF_QUOTES
import com.crossbowffs.remotepreferences.RemotePreferenceProvider
import com.yubyf.datastore.DataStorePreferences.Companion.getDataStorePreferences
import com.yubyf.quotelockx.BuildConfig

class PreferenceProvider : RemotePreferenceProvider(AUTHORITY, arrayOf(PREF_COMMON, PREF_QUOTES)) {

    override fun getSharedPreferences(context: Context, prefFileName: String): SharedPreferences {
        return context.getDataStorePreferences(prefFileName)
    }

    override fun checkAccess(prefName: String, prefKey: String, write: Boolean): Boolean {
        return !write
    }

    override fun query(
        uri: Uri,
        projection: Array<String>?,
        selection: String?,
        selectionArgs: Array<String>?,
        sortOrder: String?,
    ): Cursor? {
        return super.query(uri, projection, selection, selectionArgs, sortOrder)
    }

    companion object {
        const val AUTHORITY = BuildConfig.APPLICATION_ID + ".preferences"
    }
}